code/modules/antagonists/changeling/powers/tiny_prick.dm

```diff
@@ -99,17 +99,32 @@
 /datum/action/changeling/sting/transformation/sting_action(mob/user, mob/target)
 	log_combat(user, target, "stung", "transformation sting", " new identity is '[selected_dna.dna.real_name]'")
 	var/datum/dna/NewDNA = selected_dna.dna
+
+
 	if(ismonkey(target))
 		to_chat(user, "<span class='notice'>Our genes cry out as we sting [target.name]!</span>")
 
 	var/mob/living/carbon/C = target
 	. = TRUE
 	if(istype(C))
-		C.real_name = NewDNA.real_name
+		var/datum/dna/OldDNA = new /datum/dna
```

Comment:
```suggestion
		var/datum/dna/old_dna = new /datum/dna
```

Reply:
On second thought do you really need to initialize this DNA if you are just doing to copy it in next line?

Reply:
Yes, for a deep copy. Since old_dna = target, when I change target's DNA it will stay the same since the target dna = new dna but also old dna